src/servers/ZoneServer2016/zoneserver.ts

```diff
@@ -2481,13 +2481,29 @@ export class ZoneServer2016 extends EventEmitter {
 
     if (client.vehicle.mountedVehicle) {
       const vehicle = this._vehicles[client.vehicle.mountedVehicle],
-        container = vehicle?.getContainer();
+      container = vehicle?.getContainer(),
+      occupantsCount = vehicle.getPassengerList().length,
+      lockedState = vehicle.isLocked;
+
       if (vehicle && container) {
         container.items = {
           ...container.items,
           ...client.character.getDeathItems(this)
         };
       }
+    
+      // Check if character is dead
+      if (!client.character.isAlive) {
+        // Wait untill killCharacter is finished for dismountVehicle (if not, there will be no option for the dead occupant to respawn)
+        this.once("killCharacterComplete", (client) => {
+          this.dismountVehicle(client);
+
+          if (occupantsCount > 1 && lockedState) {
```

Comment:
I'm wondering since occupantsCount is copied prior dismount, currently since all of this func and its calls are synchronous it create no issues, but if at some point we change that, we might have an issue there, so maybe get that data directly it the event. Using an event is a good idea, i would have used a microtask but yeah if we change anything to async it would have create issue. Tell me what you think :)


```
Welcome to Node.js v23.3.0.
Type ".help" for more information.
> const arr = [1,1]
undefined
> arr.length
2
> const b = arr.length
undefined
> arr.push(1)
3
> arr.length
3
> b
2
> 
```

Reply:
also it's kinda the same for the locking state

Reply:
> @GumpyH1Z1 since you're working on a fork i can't do these changes myself btw

Let me push these changes into a branch of your repo.

Reply:
@GumpyH1Z1  you can push to your branch tho just updt this pr 🤠